core/src/main/scala/cats/data/OneAnd.scala

```diff
@@ -123,8 +123,21 @@ final case class OneAnd[F[_], A](head: A, tail: F[A]) {
     s"OneAnd(${A.show(head)}, ${FA.show(tail)})"
 }
 
+private[data] trait OneAndBinCompat0[F[_], A] {
```

Comment:
[question] Is this the right way to pass the MiMa check for a `case class` modification?

I saw https://github.com/typelevel/cats/pull/3997#issuecomment-922590054 and thought that this was the (only) way, but I could not find any other place that defines a similar bincompat `trait` (other than [`ValidatedFunctionsBinCompat0`](https://github.com/typelevel/cats/blob/e4086ed60e801b87ac8a47031ad6561eef689c8e/core/src/main/scala/cats/data/Validated.scala#L843), which patches the companion rather than the `Validated` class itself).

Reply:
can we make this sealed? I think so.